src/field_map_change.c

```diff
@@ -607,30 +606,30 @@ void FieldSystem_StartLoadMapFromErrorTask(FieldSystem *fieldSystem)
         }
     }
 
-    v1 = Heap_AllocFromHeapAtEnd(11, sizeof(UnkStruct_02053718));
-    v1->unk_00 = 0;
+    errorData = Heap_AllocFromHeapAtEnd(11, sizeof(MapChangeFromErrorData));
+    errorData->finishedFlag = FALSE;
 
-    Location_Set(&v1->unk_04, 466, -1, 8, 14, 0);
-    fieldSystem->unk_70 = 2;
-    FieldTask_Set(fieldSystem, FieldTask_LoadMapFromError, v1);
+    Location_Set(&errorData->location, 466, -1, 8, 14, 0);
+    fieldSystem->mapLoadType = MAP_LOAD_TYPE_UNION;
+    FieldTask_Set(fieldSystem, FieldTask_LoadMapFromError, errorData);
 }
 
 static BOOL FieldTask_ChangeMap(TaskManager *taskMan)
 {
     FieldSystem *fieldSystem = TaskManager_FieldSystem(taskMan);
-    UnkStruct_02053900 *v1 = TaskManager_Environment(taskMan);
-    Location *location = &v1->unk_04;
+    MapChangeData *mapChangeData = TaskManager_Environment(taskMan);
+    Location *location = &mapChangeData->nextLocation;
 
-    switch (v1->unk_00) {
+    switch (mapChangeData->state) {
     case 0:
         Sound_PlayEffect(1539);
         Sound_TryFadeInBGM(fieldSystem, location->mapId);
         sub_02055974(taskMan);
-        (v1->unk_00)++;
+        (mapChangeData->state)++;
```

Comment:
**polish**: Remove unnecessary parens (applies to other instances in this file).